### DISCUSSION TITLE
lockin / migrating away?

### DISCUSSION THREAD

**Discussion by dcsan** [pocketbase/pocketbase#4323] [Q&A]
I love the look of pocketbase for my own (side) projects, but I'm about to embark on something for a client where they will want to deploy an app on their own infra, probably on more conventional postgres etc stack. also concerns about scale - tho its probably more worries than reality.

how portable is code built for pocketbase? 
Could I use an existing ORMs server side, so that they can switch out for eg postgres later?
what about the client side realtime libs?

All of the authentication stuff i guess is very custom but I probably wouldn't use that.

sorry for the odd question!

**Comment by ganigeorgiev** [comment]
> how portable is code built for pocketbase?

Only the collections data can be migrated eventually to another database (_you still will need to transform the SQLite tables to the desired db format_).

The Web APIs, API rules, the exposed Go/JS interfaces, etc. are all specific to PocketBase and cannot be ported without rewrite.

> also concerns about scale ...

PocketBase can scale only vertical, aka. on a single server.
If that is not suitable for your use case, then you'll be better with something else.
For general benchmarks you can explore https://github.com/pocketbase/benchmarks.
I recently also shared some numbers regarding the db size and server cost from my other open source project that is using PocketBase in https://github.com/pocketbase/pocketbase/discussions/4254.

> but I'm about to embark on something for a client...

**Keep in mind that PocketBase hasn't reached v1.0.0 yet and it is not recommended for production critical applications unless you are fine with reading the [CHANGELOG](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md) and applying some manual migration steps from time to time.**
